src/compiler/crystal/types.cr

```diff
@@ -2918,7 +2913,7 @@ module Crystal
     def instantiate(type_vars)
       types = type_vars.map do |type_var|
         unless type_var.is_a?(Type)
-          type_var.raise "argument to Proc must be a type, not #{type_var}"
+          type_var.raise "argument to Union must be a type, not #{type_var}"
```

Comment:
If this error message was wrong it seems we are lacking some spec, right? Can it be added in this PR?

Reply:
Thanks!